### ISSUE TITLE
ref-objects component apply object to namespace different from app namespace causes CollectHealthStatus failed

### ISSUE THREAD

**Issue by jiangshantao-dbg** [kubevela/kubevela#4376]
**Describe the bug**
when i add ref-objects component to the application with an object which namespace different from application, the component apply success but health status failed.

i check the source code and find out that when CollectHealthStatus vela use application's namespace to get object instead of the ref-object's namespace. so the object is not found. 

error message as follows:
```
    message:step deploy: run step(provider=multicluster,do=deploy): Found 1 errors. [(error encountered in cluster local: CollectHealthStatus: app=first-vela-app-pre-bug, comp=configmap, evaluate workload status message error: get template context: failed to get obj first-vela-app-pre with gvk /v1, Kind=ConfigMap : configmaps "first-vela-app-pre" not found)]
```

**To Reproduce**
```
apiVersion: core.oam.dev/v1beta1
kind: Application
metadata:
  name: first-vela-app-pre-bug
  namespace: dev-test
spec:
  components:
  - type: ref-objects
    name: configmap
    properties:
      objects:
        - resource: configmap
          name: first-vela-app-pre
          namespace: default
          cluster: local
  policies:
  - name: topology-local
    properties:
      clusters:
      - local
    type: topology
---
apiVersion: v1
kind: ConfigMap
metadata:
  name: first-vela-app-pre
  namespace: default
```

**Expected behavior**
ref-objects Component could be used to apply objects which namespace is different from the application.

**Screenshots**

**KubeVela Version**

CLI Version: v1.4.3
Core Version: 1.4.3
GitRevision: git-5e3ab73
GolangVersion: go1.17.11

**Cluster information**

Client Version: version.Info{Major:"1", Minor:"19", GitVersion:"v1.19.7", GitCommit:"1dd5338295409edcfff11505e7bb246f0d325d15", GitTreeState:"clean", BuildDate:"2021-01-13T13:23:52Z", GoVersion:"go1.15.5", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"19", GitVersion:"v1.19.7", GitCommit:"1dd5338295409edcfff11505e7bb246f0d325d15", GitTreeState:"clean", BuildDate:"2021-01-13T13:15:20Z", GoVersion:"go1.15.5", Compiler:"gc", Platform:"linux/amd64"}

**Additional context**

**Comment by Somefive** [comment]
The bug you describe is solved in https://github.com/kubevela/kubevela/pull/4349. The next release version will fix them.

BTW, the topology policy you use should add `namespace: "dev-test"` because you are referencing an object from another namespace. If you do not add this flag, the application will try to dispatch the configmap into the original namespace ("default" in your case).

**Comment by Somefive** [comment]
Feel free to reopen this issue if you find further bugs.

**Comment by jiangshantao-dbg** [comment]
> The bug you describe is solved in #4349. The next release version will fix them.
> 
> BTW, the topology policy you use should add `namespace: "dev-test"` because you are referencing an object from another namespace. If you do not add this flag, the application will try to dispatch the configmap into the original namespace ("default" in your case).

thanks for reply. what i want is to just apply one of the ref-objects to another namespace "dev-test" and others stay the same in "default" namespace. this fix may not help?  topology policy may cause all objects to apply to the specified namespace?

**Comment by jiangshantao-dbg** [comment]
> > The bug you describe is solved in #4349. The next release version will fix them.
> > BTW, the topology policy you use should add `namespace: "dev-test"` because you are referencing an object from another namespace. If you do not add this flag, the application will try to dispatch the configmap into the original namespace ("default" in your case).
> 
> thanks for reply. what i want is to just apply one of the ref-objects to another namespace "default" and others stay the same in "dev-test" namespace. this fix may not help? topology policy may cause all objects to apply to the specified namespace?

i have try this fix, and i can help. i don't need to specify the namespace in topology policy. thank you.
